### PR TITLE
Add support for TypeScript 3

### LINKED ISSUE #686: Support for TypeScript 3
The library has TypeScript peer dependency:
`"typescript": "^2.2"`

This gives a warning with TypeScript v3+.
`warning "typestrict > tslint-sonarts@1.7.0" has incorrect peer dependency "typescript@^2.2".`

It still seems to work fine, perhaps the peer dependency could just be updated?

### PR DESCRIPTION
**Pull request by andrea-guarino-sonarsource**
Fixes #686

### CODE REVIEW

File: sonarts-core/tests/cfg/__snapshots__/cfg.test.ts.snap [hunk 1]
```diff
@@ -4478,34 +4478,6 @@ Object {
       "id": 1,
       "label": "START
 a
-a[]
```
Comment (m-g-sonar) [inline sonarts-core/tests/cfg/__snapshots__/cfg.test.ts.snap]: Why dropping this cases?

File: sonarts-core/src/rules/noMisspelledOperatorRule.ts [hunk 2]
```diff
@@ -55,6 +55,7 @@ class Visitor extends SonarRuleVisitor {
         const unaryExpression = expression as ts.PrefixUnaryExpression;
         const unaryOperator = unaryExpression.getFirstToken();
         if (
+          unaryOperator &&
```
Comment (andrea-guarino-sonarsource) [inline sonarts-core/src/rules/noMisspelledOperatorRule.ts]: Use findChild on appropriate unary operators

File: sonarts-core/src/utils/navigation.ts [hunk 3]
```diff
@@ -23,7 +23,13 @@ import * as nodes from "./nodes";
 const Kind = ts.SyntaxKind;
 
 export function keyword(node: ts.BreakOrContinueStatement | ts.ThrowStatement | ts.ReturnStatement): ts.Node {
-  return node.getFirstToken();
+  return findChild(
```
Comment (andrea-guarino-sonarsource) [inline sonarts-core/src/utils/navigation.ts]: inline this function

**Comment by andrea-guarino-sonarsource** [comment]
As discussed during standup, merging this PR.

**Pull request merged**